src/PSRule.Rules.Azure/rules/Azure.Redis.Rule.ps1

```diff
@@ -128,6 +128,12 @@ Rule 'Azure.Redis.FirewallIPRange' -Ref 'AZR-000300' -Type 'Microsoft.Cache/redi
     
 }
 
+# Synopsis: Azure Cache for Redis should use the latest supported version of Redis.
+Rule 'Azure.Redis.Version' -Ref 'AZR-000347' -Type 'Microsoft.Cache/redis' -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    $Assert.Match($TargetObject, 'properties.redisVersion', '^(6\.0\.)|(latest)').
```

Comment:
Redis version can be `6` or `6.x` or `6.x.x`.

Using `$Assert.Version` may be another option however since `latest` is also the default we need to.

- Pass with `latest` or
- Pass if the `properties.redisVersion` property is not specified.

Reply:
I'll adjust.

Reply:
`latest` isn't a valid semver expression so you could add this support by OR `$Assert.HasDefaultValue`.

I think a version constraint of `>=6` is fine to meet the requirements of the rule. The specific version will change over time.